scripts/weaver_snapshot_process.py

```
@@ -45,6 +47,38 @@
+    so whatever holds the port could send the request to a server of its
+    choosing — and the readiness poll or the ownership check would then be
+    trusting that server's content instead. A redirect is therefore proof
+    enough that the thing answering is not ``http-server`` serving this tree.
+    """
+
+    def redirect_request(  # noqa: PLR0913 - the base class fixed this signature
+        self,
+        req: urllib.request.Request,
+        fp: typ.IO[bytes],
+        code: int,
+        msg: str,
```

Ruff (0.16.2)

[warning] Unused method argument: `msg`

(ARG002)